### ISSUE TITLE
Read in HDF-EOS5 (.he5) validation data files

### ISSUE THREAD

**Issue by AliciaBentley-NOAA** [dtcenter/MET#2328]
## Describe the New Feature ##
A future version of MET will ideally be able to read in NASA's OMI dataset in order to verify Total Column Ozone. This dataset is in HDF-EOS5 (.he5) format. 

### Acceptance Testing ###
**Data on WCOSS2 are downloaded from NASA, so the data are the same**
(On WCOSS2)
/lfs/h1/ops/dev/dcom/YYYYMMDD/validation_data/aq/omi/omi_aura_${YYYY}m${mmdd}_v883-${YYYY}m${mmdd}t${HHMMSS}.he5
(From NASA)
https://omisips1.omisips.eosdis.nasa.gov/outgoing/OMTO3e/OMI-Aura_L3-OMTO3e_${YYYY}m${mmdd}_v883-${YYYY}m${mmdd}t${HHMMSS}.he5
-A test of functionality could involve a small use case using the OMI dataset for Total Column Ozone. This parameter can be found in GFS and GEFS forecasts. 

### Time Estimate ###
I'm unsure how much work would be required to do this. 
*Issues should represent approximately 1 to 3 days of work.*

### Sub-Issues ###
Consider breaking the new feature down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
*List relevant project deadlines here or state NONE.*

### Funding Source ###
*Define the source of funding and account keys here or state NONE.*

### Assignee ###
- [ ] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [ ] Select **component(s)**
- [ ] Select **priority**
- [ ] Select **requestor(s)**

### Projects and Milestone ###
- [ ] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED PROJECT ASSIGNMENT** label
- [ ] Select **Milestone** as the next official version or **Future Versions**

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [ ] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdataio](https://github.com/dtcenter/METdataio/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## New Feature Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Linked issues**
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

**Comment by JohnHalleyGotway** [comment]
Recommend considering this feature request in the larger context of the libraries upon which MET already depends.

As of version 11.0.0, MET depends on:
- NetCDF4 and HDF5 (since NC4 depends on HDF5)
- HDF4 and hdfeos version 2

The `lidar2nc` requires the HDF4 library.
The `modis_regrid` tools requires the very old hdfeos version 2 library.

Adding support for this HDF-EOS5 data will like cause confusion/conflicts with these tools. Recommend that part of this work be to revisit these tools and consider:
- Reimplementing them with the newer HDF5 and HDF-EOS5 libraries.
- Removing them from MET if they are no longer needed and/or reimplementing them is too difficult or time consuming.
